app/Jobs/DeleteWikiFinalizeJob.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Jobs;
+use App\Wiki;
+use App\WikiSetting;
+use App\WikiManager;
+use App\QueryserviceNamespace;
+use App\WikiDb;
+use Traversable;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Facades\Storage;
+use App\Http\Controllers\WikiLogoController;
+use Illuminate\Contracts\Filesystem\Cloud;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+
+class DeleteWikiFinalizeJob extends Job implements ShouldBeUnique
+{
+    private $wikiId;
+
+    /**
+     * @return void
+     */
+    public function __construct( $wikiId )
+    {
+        $this->wikiId = $wikiId;
+    }
+
+    /**
+     * The unique ID of the job.
+     *
+     * @return string
+     */
+    public function uniqueId()
+    {
+        return strval($this->wikiId);
+    }
+
+    /**
+     * @return void
+     */
+    public function handle()
+    {
+        $wiki = Wiki::withTrashed()->where('id', $this->wikiId )->first();
+
+
+        if( !$wiki ) {
+            $this->fail(new \RuntimeException("Wiki not found for {$this->wikiId}"));
+            return;
+        }
+
+        if( !$wiki->deleted_at ) {
+            $this->fail(new \RuntimeException("Wiki {$this->wikiId} is not deleted, but job got dispatched."));
+            return;
+        }
+
+        $qsNamespace = $wiki->wikiQueryserviceNamespace()->first();
+        $wikiDB = $wiki->wikiDb()->first();
+
+
+        if( $wikiDB || $qsNamespace ) {
+            $this->fail(new \RuntimeException("There are still resources allocated."));
+            return;
+        }
+
+        if ( !$this->deleteSiteDirectory( $wiki->id) ) {
+            $this->fail(new \RuntimeException("Failed deleting logos directory."));
+            return;
+        }
+
```

Comment:
It would be nice to check if elasticsearch index is also deleted here, before we delete the wiki_id which is the reference to it.
I imagine this would look similar to the check in `app/Jobs/ElasticSearchIndexDelete.php` right now to see if indexes exist?

Reply:
added in https://github.com/wbstack/api/pull/158/commits/d8ce639e38f1a1b4fe9eae59ece05efa1fc1b984